ios/FluentUI/Badge Field/BadgeView.swift

```diff
@@ -187,11 +137,148 @@ open class BadgeView: UIView {
+            }
+        }
+        set {
+            if selectedLabelTextColor != newValue {
+                _selectedLabelTextColor = newValue
+                updateColors()
+            }
+        }
+    }
+    
+    private var _disabledLabelTextColor: UIColor?
+    open var disabledLabelTextColor: UIColor? {
+        get {
+            if let customDisabledLabelTextColor = _disabledLabelTextColor {
+                return customDisabledLabelTextColor
+            }
+            return style == .default ? Colors.Badge.textDisabled : (isActive ? self.selectedLabelTextColor : self.labelTextColor)
+        }
+        set {
+            if disabledBackgroundColor != newValue {
+                _disabledLabelTextColor = newValue
+                updateColors()
+            }
+        }
+    }
+    
+    private var _backgroundColor: UIColor?
+    open override var backgroundColor: UIColor? {
+        get {
+            if let customBackgroundColor = _backgroundColor {
+                return customBackgroundColor
+            }
+            
+            switch style {
+                case .default:
+                    if let window = window {
+                        return Colors.primaryTint40(for: window)
+                    }
+                case .warning:
+                    return Colors.Badge.backgroundWarning
+                case .error:
+                    return Colors.Badge.backgroundError
+            }
+            return nil
+        }
+        set {
+            if backgroundColor != newValue {
+                _backgroundColor = newValue
+                updateColors()
+            }
+        }
+    }
+    
+    private var _selectedBackgroundColor: UIColor?
+    open var selectedBackgroundColor: UIColor? {
+        get {
+            if let customSelectedBackgroundColor = _selectedBackgroundColor {
+                return customSelectedBackgroundColor
+            }
+            
+            switch style {
+                case .default:
+                    if let window = window {
+                        return Colors.primary(for: window)
+                    }
+                case .warning:
+                    return Colors.Badge.backgroundWarningSelected
+                case .error:
+                    return Colors.Badge.backgroundErrorSelected
+            }
+            return nil
+        }
+        set {
+            if selectedBackgroundColor != newValue {
+                _selectedBackgroundColor = newValue
+                updateColors()
+            }
+        }
+    }
+    
+    private var _disabledBackgroundColor: UIColor?
+    open var disabledBackgroundColor: UIColor? {
+        get {
+            if let customDisabledBackgroundColor = _disabledBackgroundColor {
+                return customDisabledBackgroundColor
+            }
+            
+            return style == .default ? Colors.Badge.backgroundDisabled : (isActive ? self.selectedBackgroundColor : self.backgroundColor)
```

Comment:
Same here... The disabled background color should either return the custom set or Colors.Badge.backgroundDisabled.

Reply:
It first checks for custom colors if none found it looks for presets value but presets value are not defined for `disabledBackground`  or `disabledLabelText` (other than default `style`).

Reply:
I see.
Another question I have is: wouldn't the check be on isSelected?

return style == .default ? Colors.Badge.backgroundDisabled : (**isSelected** ? self.selectedBackgroundColor : self.backgroundColor)

Reply:
Thanks Rafael for catching this. You are right about `isSelected`. I will update this. It's also strange as it seems to be I missed 4 test cases. I will add some testing to make sure I have an exhaustive test cases.

Reply:
I have updated this, I will create another PR for adding snapshot testing after a doc review.